src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -53,15 +53,15 @@
    * Get the historical price for each symbol passed in for the date passed in.
    *
    * @param symbol list of symbols to get historical price for.
-   * @param date (formatted YYYYMMDD) the date for which you want the price on.
+   * @param dateOrRange the date (formatted YYYYMMDD) or the range of time  (ex. "5m", "ytd" ).
    * @return a List of IexHistoricalPrices objects for the given symbols.
    */
   @GetMapping(value = "${mvc.iex.getHistoricalPricesPath}", produces = {
       MediaType.APPLICATION_JSON_VALUE})
   public List<IexHistoricalPrices> getHistoricalPrices(
       @RequestParam final String symbol,
-      @RequestParam final String date) {
-    return iexService.getHistoricalPricesForSymbol(symbol, date);
+      @RequestParam final String dateOrRange) {
+    return iexService.getHistoricalPricesForSymbol(symbol, dateOrRange);
```

Comment:
Why not use @RequestParam(optional=true) and have a date and a range variable, instead of stuffing both in one? Also how would you tell in this model if its a date or a range?

Reply:
I changed this in the code - however; I had to make some "arbitrary" decisions regarding priority. If neither of the optional parameters were given, I defaulted to using "1m" for the range - the same as the Iex API does. If both are included, I arbitrarily chose to give precedent to the date over the range. I can easily change the logic here if needed

Reply:
Sorry for the delay - handling the null inputs when not including the required=false parameters took me some time to figure out. Should be set now.